great_expectations/cli/batch_request.py

```diff
@@ -174,12 +175,11 @@ def select_data_connector_name(
     return data_connector_name
 
 
-def _get_data_asset_name_from_data_connector(
+def get_data_asset_name_from_data_connector(
```

Comment:
why was this promoted to public? ... and where is the docstring? ;)

Reply:
It's responsibilities have grown considerably with the introduction of pagination and search so I thought it made sense to signify that through access modifiers.

It also makes testing that one component a lot easier (but I do realize that access modification to make testing is not the best practice).

I think the first point is still valid but let me know if you disagree and we can determine an alternative approach.

Reply:
When making something public, the question isn't about the amount of responsibility, it's about whether or not you expect other people to call it.

Reply:
per discussion, keep this as a private function but still leaving docstring and function-specific tests